aquacomputer_d5next.c

```diff
@@ -474,14 +512,43 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 	switch (type) {
 	case hwmon_pwm:
 		switch (attr) {
+		case hwmon_pwm_enable:
+			if (priv->fan_ctrl_offsets != NULL) {
+				switch (priv->kind) {
+				case d5next:
+					if (val < 0 || val > 3)
+						return -EINVAL;
+					break;
+				case quadro:
+				case octo:
+					if (val < 0 || val > priv->num_fans + 3)
+						return -EINVAL;
+					if (val == channel + 4)
```

Comment:
Please add a comment that this is checked so it does not follow itself, it's not immediately clear why `+ 4`.